### ISSUE TITLE
Config framework

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#753]
After reviewing the options, I am currently implementing `confuse` based on the discussion linked below.  Steps involved roughly outlined. I'm finishing up some documentation now.

I plan to revert default `nufft` to `finufft` and change the gpu ci job to use the new config in a small followup PR.

- [x] Remove unused/unneeded config vars. This minimized conversion
- [x] Replace legacy config and config parser with `confuse`
    - [x] Add confuse to the package
    - [x] Implemented new config as `cconfig`, leaving old in place
    - [x] Replace the config var uses with ccconfig.
    - [x] Remove legacy config entirely
    - [x] String replace `cconfig`  with `config`
    - [x] Add some logging for the config at startup in case we need it late.
    - [x] Add confuse config to Python package
- [x] Add documentation/tutorial about overrides.

This PR should close #504 and unblock #712 among other things.

### Discussed in https://github.com/ComputationalCryoEM/ASPIRE-Python/discussions/698

<div type='discussions-op-text'>

<sup>Originally posted by **chris-langfield** September 21, 2022</sup>
Some things we want to be able to do with a config system after talking with @garrettwrong:
1 - Automatically detect config file in package directory. In particular, we want automatic path searching. For instance, first look for a `config.ini` in the working directory, then in the home directory, finally defaulting to a config file shipped with the package.
2 - A simple file format. We currently use `.ini` along with the standard library's `configparser`. Other options are YAML, TOML, or even `.py` 
3 - We may want an environment variable set to path to config file?
4-  We want to be able to ship a GPU config file ready to go

Some notes on the options out there:
- [`configparser`](https://docs.python.org/3/library/configparser.html): From what I can tell, this does not implement 1) above. It could be quite involved to write a cross-platform path search
- [`confuse`](https://confuse.readthedocs.io/en/latest/): This is for YAML. We have heard good things from RC's Python expert
- [`dynaconf`](https://dynaconf.readthedocs.io/en/docs_223/): This is popular on Reddit and StackOverflow threads, but looks like it's probably overkill. Also seems to have partly dropped 3.7 support
- [`config-path`](https://github.com/barry-scott/config-path) This might do what we want as far as cross-platform path search. It hasn't been updated in a while but looks official and is marked as "production/stable" on PyPI.

`configparser` + `config-path` + INI's is jumping out at me as the simplest solution to look into first.
</div>

**Comment by garrettwrong** [comment]
Also unblocks #743 .

**Comment by garrettwrong** [comment]
Closed by #749
